projects/auth0-angular/src/lib/auth.service.spec.ts

```diff
@@ -203,6 +203,22 @@ describe('AuthService', () => {
         done();
       });
     });
+
+    it('should record errors in the error$ observable', (done) => {
+      (auth0Client.handleRedirectCallback as jasmine.Spy).and.throwError(
+        new Error('An error occurred')
+      );
+
+      const localService = createService();
+
+      loaded(localService).subscribe(() => {
+        localService.error$.subscribe((error: Error) => {
```

Comment:
Thinking this from the developer view, matching against an "error message" is not useful (messages can change). How generic are these `Error` types? Shouldn't we provide at least a type that covers error code + error description? so the code value can be used to match.

Reply:
Here I just want to show that if an error is raised when calling `handleRedirectCallback` then it is handled and emitted through the `error$` observable. I don't care what the message is as I created the message inside the same test. However, I have refactored it to check the actual error object instead of the message.

In practise, this error will be whatever is thrown by the underlying SDK, which is [normalized as you suggest](https://github.com/auth0/auth0-spa-js/blob/master/src/Auth0Client.ts#L470-L479).